pgcli/main.py

```diff
@@ -984,13 +984,24 @@ def format_arrays(data, headers, **_):
 
     if cur:
         headers = [case_function(utf8tounicode(x)) for x in headers]
-        rows = list(cur)
-        formatted = formatter.format_output(rows, headers, **output_kwargs)
+        if max_width is not None:
+            cur = list(cur)
+        column_types = None
+        if hasattr(cur, 'description'):
+            column_types = []
+            for d in cur.description:
+                if d[1] == 23:
+                    column_types.append(int)
+                elif d[1] == 1700:
+                    column_types.append(float)
```

Comment:
`psycopg2.extensions.cursor` has a method `cast()` that can be used with oid's. I didn't find a method to get the type itself though, but with those magic numbers you might also miss some types.